### PR TITLE
Fix encoding/decoding for UTF-8 stomp commands and headers

### PR DESCRIPTION
**Pull request by amizurov**
Motivation:

According STOMP spec (https://stomp.github.io/stomp-specification-1.2.html#Value_Encoding) we have to encode and decode commands and headers to UTF-8

Modification:

Provide ability for StompSubframeDecoder and StompSubframeEncoder work with UTF-8

### CODE REVIEW

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java [hunk 1]
```diff
@@ -94,8 +88,8 @@ public StompSubframeDecoder(int maxLineLength, int maxChunkSize, boolean validat
         checkPositive(maxLineLength, "maxLineLength");
         checkPositive(maxChunkSize, "maxChunkSize");
         this.maxChunkSize = maxChunkSize;
-        this.maxLineLength = maxLineLength;
-        this.validateHeaders = validateHeaders;
+        commandParser = new Utf8LineParser(new AppendableCharSequence(16), maxLineLength);
+        headerParser = new HeaderParser(new AppendableCharSequence(123), maxLineLength, validateHeaders);
```
Comment (normanmaurer) [inline codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java]: why 123 ? (just curious)

File: codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java [hunk 2]
```diff
@@ -252,75 +236,140 @@ private static void skipControlCharacters(ByteBuf buffer) {
...
-        for (;;) {
-            byte nextByte = buffer.readByte();
+    private void resetDecoder() {
+        checkpoint(State.SKIP_CONTROL_CHARACTERS);
+        contentLength = -1;
+        alreadyReadChunkSize = 0;
+        lastContent = null;
+    }
+
+    static class Utf8LineParser implements ByteProcessor {
```
Comment (normanmaurer) [inline codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java]: private ?

File: codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java [hunk 3]
```diff
@@ -252,75 +236,140 @@ private static void skipControlCharacters(ByteBuf buffer) {
...
+
+            if (nextByte == StompConstants.LF) {
+                return false;
+            }
+
+            if (++lineLength > maxLineLength) {
+                throw new TooLongFrameException("An STOMP line is larger than " + maxLineLength + " bytes.");
+            }
+
+            if (nextRead) {
```
Comment (normanmaurer) [inline codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java]: can you add some comments here to explain all of this ?

File: codec-stomp/src/test/java/io/netty/handler/codec/stomp/StompSubframeEncoderTest.java [hunk 4]
```diff
@@ -63,4 +67,23 @@ public void testFrameAndContentEncoding() {
...
+        StompHeaders incoming = frame.headers();
+        incoming.set(StompHeaders.DESTINATION, "/queue/№11±♛нетти♕");
+        incoming.set(StompHeaders.CONTENT_TYPE, AsciiString.of("text/plain"));
+
+        channel.writeOutbound(frame);
+
+        ByteBuf headers = channel.readOutbound();
+        ByteBuf content = channel.readOutbound();
+        ByteBuf fullFrame = new CompositeByteBuf(ByteBufAllocator.DEFAULT, false, 2)
+                .addComponent(true, headers).addComponent(true, content);
```
Comment (normanmaurer) [inline codec-stomp/src/test/java/io/netty/handler/codec/stomp/StompSubframeEncoderTest.java]: nit: Use `Unpooled.wrappedBuffer(...)`

File: codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java (hunk 1 above)
Comment (amizurov) [reply codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java]: oh, typo 128

File: codec-stomp/src/test/java/io/netty/handler/codec/stomp/StompSubframeEncoderTest.java (hunk 4 above)
Comment (amizurov) [reply codec-stomp/src/test/java/io/netty/handler/codec/stomp/StompSubframeEncoderTest.java]: thanks, done

File: codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java (hunk 3 above)
Comment (amizurov) [reply codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java]: added some comments, if it's not clear ping me

File: codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java (hunk 2 above)
Comment (amizurov) [reply codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java]: done

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@amizurov thanks a lot!

**Pull request merged**